### ISSUE TITLE
ValueError: Invalid scipy.LowLevelCallable signature on Windows

### ISSUE THREAD

**Issue by j08lue** [jni/llc-tools#3]
Thanks a lot for posting these recipes. I am trying to get your `jit_filter_function` decorator to work, but I get the same error as in [`one of the comments on your blog post`](https://ilovesymposia.com/2017/03/15/prettier-lowlevelcallables-with-numba-jit-and-decorators/#comment-9615):
```
ValueError: Invalid scipy.LowLevelCallable signature "long (double *, longlong, double *, void *)". Expected one of: ['int (double *, intptr_t, double *, void *)', 'int (double *, npy_intp, double *, void *)', 'int (double *, long long, double *, void *)']
```

This is with Python 3.6 on **Windows** and
```python
>>> numba.__version__
'0.35.0+10.g143f70e'
>>> scipy.__version__
'0.19.1'
```

I have tried to modify this line
https://github.com/jni/llc-tools/blob/576d70cc07244b727173bbca1e0f8029e76e5a8a/llc/ndimage.py#L11
in [any way I could think of](http://numba.pydata.org/numba-doc/0.12.2/tutorial_types.html#supported-types-in-numba), but am unable to produce a signature that `scipy.LowLevelCallable` likes. What am I missing?

**Comment by j08lue** [comment]
Here is a minimal example:

```python
import llc
import numpy as np
import scipy.ndimage as ndi

@llc.jit_filter_function
def fmin(values):
    result = np.inf
    for v in values:
        if v < result:
            result = v
    return result

ndi.generic_filter(np.random.random((20, 30)), fmin, 3)
```

And it works under Linux. 

So the issue seems to be Windows thing, possibly related to https://github.com/numba/numba/issues/1836.

**Comment by jni** [comment]
@j08lue Are you saying that the `fmin` example, doesn't work in Windows? =\ And the error is as above?

**Comment by j08lue** [comment]
Sadly so. Here is the complete story:
https://gist.github.com/j08lue/ddfade5f1a08bedb06d3ff4d5cebb568

**Comment by dionhaefner** [comment]
I have [raised an issue](https://github.com/numba/numba/issues/2578) in the Numba repository.

**Comment by jni** [comment]
@j08lue thank you for the report and @dionhaefner thank you for raising with Numba! From the report there, it seems that there is actually no way to make this work on Windows currently, right?

**Comment by dionhaefner** [comment]
Yes, it seems like it.

**Comment by MartinSJRogers** [comment]
Hi @jni I tried your code as well as thought it incredibly useful, but I came up with the same error message mentioned above:
ValueError: Invalid scipy.LowLevelCallable signature "long (double *, longlong, double *, void *)". Expected one of: ['int (double *, intptr_t, double *, void *)', 'int (double *, npy_intp, double *, void *)', 'int (double *, long long, double *, void *)']

I'm windows 10, python 3.7. Do you know if there's still no way to make this work in Windows? 

Many thanks

**Comment by trhallam** [comment]
I was able to get this to work on windows by forcing the signature argument of `LowLevelCallable`. I have to admit, this was driving me nuts this morning.

```
def jit_filter_function(filter_function):
    """Decorator for use with scipy.ndimage.generic_filter."""
    jitted_function = numba.jit(filter_function, nopython=True)

    @cfunc(intc(CPointer(float64), intp, CPointer(float64), voidptr))
    def wrapped(values_ptr, len_values, result, data):
        values = carray(values_ptr, (len_values,), dtype=float64)
        result[0] = jitted_function(values)
        return 1
    return LowLevelCallable(wrapped.ctypes, signature="int (double *, npy_intp, double *, void *)")
 ```

**Comment by jni** [comment]
@trhallam wow! That's awesome! Would you like to submit a PR to fix this in master?

**Comment by trhallam** [comment]
#11 hopefully fixes this for `generic_filter` and `generic_filter1d`.

**Comment by jni** [comment]
Closed by #11, released in 0.3.1! Thank you @trhallam! If anyone else encounters this after 0.3.1 please reopen this issue.
